app/src/main/java/fi/helsinki/opintoni/service/FeedbackService.java

```diff
@@ -68,4 +123,65 @@ public void insertFeedback(InsertFeedbackRequest request) throws Exception {
         feedback.comment = feedbackDto.comment;
         return getAllFeedback();
     }
+
+    private String createTimestamp() {
+        ZonedDateTime helsinkiNow = ZonedDateTime.now(HELSINKI_ZONE_ID);
+        return helsinkiNow.format(TIMESTAMP_FORMATTER);
+    }
+
+    private String getState(InsertFeedbackRequest request) {
+        return request.metadata.findValue("state").textValue();
+    }
+
+    private String getFaculty(InsertFeedbackRequest request) {
+        JsonNode facultyNode = request.metadata.findValue("faculty");
+        return facultyNode == null ? null : facultyNode.textValue();
+    }
+
+    private String getUserAgent(InsertFeedbackRequest request) {
+        return request.metadata.findValue("userAgent").textValue();
+    }
+
+    private Locale getLocale(InsertFeedbackRequest request) {
+        JsonNode langNode = request.metadata.findValue("lang");
+        String langCode = langNode == null ? "fi" : langNode.textValue();
+        return CODES_TO_LOCALES.get(langCode);
+    }
+
+    private String buildMessageBody(String content,
+                                    String timestamp,
+                                    String faculty,
+                                    String state,
+                                    String userAgent,
+                                    Locale locale) {
+        String timestampLine = getTimestampLine(timestamp, locale);
+        String stateLine = getStateLine(state, locale);
+        String userAgentLine = getUserAgentLine(userAgent, locale);
+        if (faculty == null) {
+            return String.join("\n", content, "", timestampLine, stateLine, userAgentLine);
```

Comment:
Tässä kannattaa melkein käyttää `StringJoiner`ia niin voi alussa lisätä siihen kaikki alun rivit, sitten katsoa `if`issä lisätäänkö faculty ja lisätä se ja `if`in jälkeen lisätä `StringJoiner`iin lopun kentät ja kutsua sille `.toString()` (nyt rivit 161 ja 163 ovat melkein identtiset)